pkg/scalers/aws_cloudwatch_scaler.go

```diff
@@ -287,19 +280,27 @@ func (c *awsCloudwatchScaler) GetMetrics(ctx context.Context, metricName string,
 
 	metric := external_metrics.ExternalMetricValue{
 		MetricName: metricName,
-		Value:      *resource.NewQuantity(int64(metricValue), resource.DecimalSI),
+		Value:      *resource.NewQuantity(metricValue, resource.DecimalSI),
 		Timestamp:  metav1.Now(),
 	}
 
 	return append([]external_metrics.ExternalMetricValue{}, metric), nil
 }
 
 func (c *awsCloudwatchScaler) GetMetricSpecForScaling(context.Context) []v2beta2.MetricSpec {
+	var metricNameSuffix string
+
+	if c.metadata.expression != "" {
+		metricNameSuffix = c.metadata.metricsName
+	} else {
+		metricNameSuffix = c.metadata.dimensionName[0]
+	}
+
 	externalMetric := &v2beta2.ExternalMetricSource{
 		Metric: v2beta2.MetricIdentifier{
-			Name: GenerateMetricNameWithIndex(c.metadata.scalerIndex, kedautil.NormalizeString(fmt.Sprintf("aws-cloudwatch-%s", c.metadata.dimensionName[0]))),
+			Name: GenerateMetricNameWithIndex(c.metadata.scalerIndex, kedautil.NormalizeString(fmt.Sprintf("aws-cloudwatch-%s", metricNameSuffix))),
```

Comment:
I am not sure if we should do this, maybe we can just use this:
`Name: GenerateMetricNameWithIndex(c.metadata.scalerIndex, kedautil.NormalizeString("aws-cloudwatch")),

Seems like that `c.metadata.metricsName` hasn't been used before, it is safe to drop it from here.
@JorTurFer WDYT?